apis/metal3.io/v1alpha1/baremetalhost_validation.go

```diff
@@ -71,6 +77,11 @@ func (host *BareMetalHost) validateChanges(old *BareMetalHost) []error {
 		errs = append(errs, fmt.Errorf("bootMACAddress can not be changed once it is set"))
 	}
 
+	if old.Status != (BareMetalHostStatus{}) && host.Annotations[StatusAnnotation] != "" {
```

Comment:
question: should this be something like

```
if old.Status != BareMetalHostStatus{} && host.Annotations[StatusAnnotation] != "" && host.Annotations[StatusAnnotation] != old.Annotations[StatusAnnotation] {
```

i.e. could this break other writes to the CR in the period from when the operator copies the status to the Status until the status annotation is removed? (I don't actually know.)

Reply:
@zaneb has a valid point, I am thinking if we can break this long if condition somehow.

Reply:
@kashifest i have included another if inside to break it the condition into two.